File: Net/TemplateMethod/Before/TemplateMethod/Tea.cs
namespace TemplateMethod
{
    using System;

    public class Tea
    {
        public void PrepareRecipe()
        {
            this.BoilWater();
            this.SteepTeaBag();
            this.PourInCup();
            this.AddLemon();
        }

        private void BoilWater()
        {
            Console.WriteLine(BeverageMessages.BoilWater);
        }

        private void SteepTeaBag()
        {
            Console.WriteLine(BeverageMessages.SteepTeaBag);
        }

        private void PourInCup()
        {
            Console.WriteLine(BeverageMessages.PourInCup);
        }

        private void AddLemon()
        {
            Console.WriteLine(BeverageMessages.AddLemon);
        }
    }
}

File: Java/TemplateMethod/Before/src/main/templatemethod/Tea.java
package templatemethod;

public class Tea {
 
	public void prepareRecipe() {
		boilWater();
		steepTeaBag();
		pourInCup();
		addLemon();
	}
 
	private void boilWater() {
		System.out.println(BeverageMessages.BoilWater);
	}
 
	private void steepTeaBag() {
		System.out.println(BeverageMessages.SteepTeaBag);
	}
 
	private void pourInCup() {
		System.out.println(BeverageMessages.PourInCup);
	}
	
	private void addLemon() {
		System.out.println(BeverageMessages.AddLemon);
	}
}
